File: Net_Core/Web.Support/Utilities/DBInformationUtility.cs
using System.Threading.Tasks;
using GrowthWare.BusinessLogic;
using GrowthWare.Framework;
using GrowthWare.Framework.Models;

namespace GrowthWare.Web.Support.Utilities;

/// <summary>
/// DBInformationUtility serves as the focal point for any web application needing to utilize the GrowthWare framework.
/// Web needs such as caching are handled here.
/// </summary>
public static class DBInformationUtility
{
    private static BDBInformation m_BDBInformation = null;

    /// <summary>
    /// New instance of the class
    /// </summary>
    /// <returns>MDBInformation</returns>
    public static async Task<MDBInformation> DBInformation()
    {
        return await(await getBusinessLogic()).GetProfile();
    }

    /// <summary>
    /// Returns the business logic object used to access the database.
    /// </summary>
    /// <returns></returns>
    private static async Task<BDBInformation> getBusinessLogic()
    {
        if (m_BDBInformation == null || ConfigSettings.CentralManagement == true)
        {
            m_BDBInformation = new(await SecurityEntityUtility.CurrentProfile());
        }
        return m_BDBInformation;
    }

    /// <summary>
    /// Updated the information in the data store
    /// </summary>
    /// <param name="profile">MDBInformation</param>
    /// <returns>bool or exception</returns>
    public static async Task<bool> UpdateProfile(MDBInformation profile)
    {
        bool mRetVal = false;
        mRetVal = await (await getBusinessLogic()).UpdateProfile(profile);
        return mRetVal;
    }
}

File: Net_Framework/VB/GrowthWare.WebSupport/Utilities/DBInformationUtility.vb
Imports GrowthWare.Framework.Model.Profiles
Imports GrowthWare.Framework.BusinessData.BusinessLogicLayer
Imports GrowthWare.Framework.Common

Namespace Utilities
    ''' <summary>
    ''' DBInformationUtility serves as the focal point for any web application needing to utilize the GrowthWare framework.
    ''' Web needs such as caching are handled here.
    ''' </summary>
    Public Class DBInformationUtility
        ''' <summary>
        ''' New instance of the class
        ''' </summary>
        ''' <returns>MDBInformation</returns>
        Public Shared Function DBInformation() As MDBInformation
            Dim mBLL As BDBInformation = New BDBInformation(SecurityEntityUtility.CurrentProfile, ConfigSettings.CentralManagement)
            Return mBLL.GetProfile
        End Function
        ''' <summary>
        ''' Updated the information in the data store
        ''' </summary>
        ''' <param name="profile">MDBInformation</param>
        ''' <returns>bool or exception</returns>
        Public Shared Function UpdateProfile(profile As MDBInformation) As Boolean
            Dim success As Boolean = False
            Dim mBLL As BDBInformation = New BDBInformation(SecurityEntityUtility.CurrentProfile, ConfigSettings.CentralManagement)
            success = mBLL.UpdateProfile(profile)
            Return success
        End Function
    End Class
End Namespace
